engine/lib/actions.php

```diff
@@ -303,10 +319,8 @@ function _elgg_csrf_token_refresh() {
 		'user_guid' => elgg_get_logged_in_user_guid(),
 	);
 
-	header("Content-Type: application/json;charset=utf-8");
-	echo json_encode($data);
-
-	return true;
+	elgg_set_http_header("Content-Type: application/json;charset=utf-8");
+	return elgg_ok_response($data);
```

Comment:
@hypeJunction here's all I changed for the token refresh. I just deleted your test (it didn't cover the new behavior), but I'll try to add the test back.